### ISSUE TITLE
Build time conflict with libevent

### ISSUE THREAD

**Issue by l2dy** [minetest/minetest#9078]
##### Minetest version

```
f2e62298a5b204045f01e399e78eb84072df967a
```

##### OS / Hardware

Operating system: macOS 10.14.6

##### Summary

When building camera.cpp, system headers are preferred to internal headers, causing build failure because `event.h` from the system is used instead if libevent is installed.

`-I/opt/local/include -I/opt/local/include/leveldb -I/opt/local/include/hiredis -I/Users/*/Build/minetest/cmakebuild/src -I/Users/*/Build/minetest/src [...]`

##### Steps to reproduce

`cmake -DENABLE_SYSTEM_JSONCPP=1 .. && make` with libevent and jsoncpp installed from MacPorts.

```
-- *** Detected Git version 5.2.0-dev-f2e62298 ***
[ 23%] Built target GenerateVersion
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/CMakeFiles/minetest.dir/build.make src/CMakeFiles/minetest.dir/depend
cd /Users/*/Build/minetest/cmakebuild && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /Users/*/Build/minetest /Users/*/Build/minetest/src /Users/*/Build/minetest/cmakebuild /Users/*/Build/minetest/cmakebuild/src /Users/*/Build/minetest/cmakebuild/src/CMakeFiles/minetest.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/CMakeFiles/minetest.dir/build.make src/CMakeFiles/minetest.dir/build
[ 23%] Building CXX object src/CMakeFiles/minetest.dir/client/camera.cpp.o
cd /Users/*/Build/minetest/cmakebuild/src && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DUSE_CMAKE_CONFIG_H -I/opt/local/include -I/opt/local/include/leveldb -I/opt/local/include/hiredis -I/Users/*/Build/minetest/cmakebuild/src -I/Users/*/Build/minetest/src -I/opt/local/include/irrlicht -I/Users/*/Build/minetest/src/Release -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenAL.framework/Headers -I/opt/local/include/luajit-2.0 -I/Users/*/Build/minetest/src/script -I/opt/local/include/freetype2  -std=c++11 -DNDEBUG -Wall   "-D__extern_always_inline=extern __always_inline" -Wsign-compare -Wall -pipe -funroll-loops -Os -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14   -std=gnu++11 -o CMakeFiles/minetest.dir/client/camera.cpp.o -c /Users/*/Build/minetest/src/client/camera.cpp
/Users/*/Build/minetest/src/client/camera.cpp:165:32: error: member access into incomplete type 'MtEventManager'
                                m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::VIEW_BOBBING_STEP));
                                                           ^
/Users/*/Build/minetest/src/client/client.h:57:7: note: forward declaration of 'MtEventManager'
class MtEventManager;
      ^
/Users/*/Build/minetest/src/client/camera.cpp:183:32: error: member access into incomplete type 'MtEventManager'
                                m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::CAMERA_PUNCH_LEFT));
                                                           ^
/Users/*/Build/minetest/src/client/client.h:57:7: note: forward declaration of 'MtEventManager'
class MtEventManager;
      ^
/Users/*/Build/minetest/src/client/camera.cpp:185:32: error: member access into incomplete type 'MtEventManager'
                                m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::CAMERA_PUNCH_RIGHT));
                                                           ^
/Users/*/Build/minetest/src/client/client.h:57:7: note: forward declaration of 'MtEventManager'
class MtEventManager;
      ^
3 errors generated.
make[2]: *** [src/CMakeFiles/minetest.dir/client/camera.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/minetest.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by sfan5** [comment]
This has worked since 2012 on every operating system Minetest supports (including macOS).
Not a bug, fix your build configuration.

**Comment by sfan5** [comment]
Where do the `-I/opt/local/include -I/opt/local/include/leveldb -I/opt/local/include/hiredis` flags come from anyway?
What does cmake output? Do these paths appear in CMakeCache.txt?

**Comment by PaddyMac** [comment]
I've encountered the same issue on FreeBSD (12.1-STABLE). It isn't surprising that the issue would occur on both OSes since MacOS borrows some code from FreeBSD. I ran into other build failures while trying to build Minetest 5.3.0 from FreeBSD ports, and then I ran into this issue while building Minetest manually.

**Comment by sfan5** [comment]
fixed by 62913b872ea1b21a5aada55ed323476fbcea61dc
